### PR TITLE
fix: regex validation issue in property pane.

### LINKED ISSUE #7485: [Bug] PropertyPane validations are not working as expected
## Description

I tried to pass a random non-URL value to Video Widget's URL and it is not showing any error.
In the code, I could see that we are validating this field using Regex.

### Steps to reproduce the behaviour:

Add a video widget and pass in any non-URL to its URL field.

https://user-images.githubusercontent.com/10436935/133488535-225af348-359b-4ae4-96a7-fa67eba8b407.mov

### Important Details

- Version: [Cloud]
- OS: [MacOS]
- Browser [chrome]
- Environment [production]

### PR DESCRIPTION
**Pull request by Kathuria**
## Description

> PropertyPane validations are not working as expected #7485

Fixes #7485 

> FIxed video url regex

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Validated on UI

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Kathuria** [comment]
@aswathkk : Please validate this PR, this is fixed now.

Validated test cases as well.

**Comment by Kathuria** [comment]
@aswathkk: Can you please review this PR and merge if looks good. In case you planned to merge later can you please add hacktoberfest-accepted label to this PR considering last day of hacktoberfest.

**Review by aswathkk — APPROVED** [review]
@Kathuria The changes look good. I've made some review comments.
I'm accepting this PR for now for hacktoberfest.

Please do address the review comments. We will merge it afterwards.

File: app/client/src/workers/validations.ts [hunk 1]
```diff
@@ -349,16 +349,11 @@ export const VALIDATORS: Record<ValidationTypes, Validator> = {
       }
     }
 
-    if (
-      config.params?.regex &&
-      isString(config.params?.regex) &&
-      !config.params?.regex.test(parsed as string)
-    ) {
+    const videoRegex = new RegExp(config.params?.regex as RegExp);
```
[the remark this answers is not in the document]
Comment (Kathuria) [reply app/client/src/workers/validations.ts]: @aswathkk : This makes sense. Incorporated all mentioned changes, please have a look.

**Review by aswathkk — APPROVED** [review]
LGTM

**Comment by aswathkk** [comment]
@Kathuria Some tests are failing. Can you merge the latest release branch into your branch and push it here?

**Comment by Kathuria** [comment]
@aswathkk : Done, Can you check now.

**Comment by aswathkk** [comment]
Test run: https://github.com/appsmithorg/appsmith/actions/runs/1406788908

**Comment by aswathkk** [comment]
Last two runs:
https://github.com/appsmithorg/appsmith/actions/runs/1407035617
https://github.com/appsmithorg/appsmith/actions/runs/1407472179

**Comment by aswathkk** [comment]
test run: https://github.com/appsmithorg/appsmith/actions/runs/1407659512

**Comment by aswathkk** [comment]
@shwetha-ramesh Once we merge #9177, the error message will be shown properly as "This value does not evaluate to type Video URL".

**Comment by aswathkk** [comment]
Closing this since the issue is not reproducible in the current release.

**Pull request closed without merging**